server/api/lib/tags/index.js

```diff
@@ -0,0 +1,62 @@
+ const { tags, tagDataRules } = require('../../constants');
+ const PlanAreaChanges = require('../../../api/model/plan_area_changes');
+
+ const  generateTagsForPlan = async (planId) => {
+    let planTags = [];
+    let dataRules = [];
+    // HOUSING
+    const housingTag = tags['דיור'];
+    // check housing by square meters
+    const housingByArea = await isTag(planId,housingTag,'housingByArea'); 
+        if (housingByArea) {
+            dataRules.push( 
+            housingByArea.created_by_data_rules
+        );
+    }
+    // check housing by area
+    const housingByUnits = await isTag(planId,housingTag,'housingByUnits');
+    if (housingByUnits) {
+        dataRules.push( 
+            housingByUnits.created_by_data_rules
+        );
+    }
+    if (housingByArea || housingByUnits)
+    planTags.push( {
+        plan_id: planId,
+        tag_id: housingTag,
+        display_score: 0, /* TODO: Add the correct display score here */
+        created_by_data_rules: `[${dataRules.toString(',')}]`
+    });
+    return planTags;
+}
+
+const isTag = async (planId, tag, rule) => {
```

Comment:
Can you move this function into housing_tag.js os something similliar?
After discussing with the team, it's best best to keep every tag in a seperate file.